src/main/scala/org/apache/spark/sql/delta/actions/actions.scala

```diff
@@ -168,6 +171,11 @@ case class RemoveFile(
 
   @JsonIgnore
   val delTimestamp: Long = deletionTimestamp.getOrElse(0L)
+
+  @JsonIgnore
+  lazy val partitionValues: Map[String, String] = PartitioningUtils.parsePathFragment(
+    path.split("/").filter(_.contains("=")).mkString("/")
+  )
```

Comment:
This isn't safe to do. See the protocol spec: https://github.com/delta-io/delta/blob/master/PROTOCOL.md#data-files

> This directory format is only used to follow existing conventions and is not required by the protocol. Actual partition values for a file must be read from the transaction log.

Reply:
I was afraid of this one... I don't think its a good design to parse partition values at all, but didn't have many options. I see 2 possibilities:
**1. add partitionValues as a case class property with `@JsonIgnore`** - we have to be sure its not a problem to read RemoveFile from log with empty partitionValues. I think it should be fine since partitionValues will be checked on newly created RemoveFiles.

**2. add partitionValues as a case class property without `@JsonIgnore`**
This requires protocol change and I don't think you want to go this way.

What do you think ?

Reply:
After another look, I don't think either is okay. The former can cause correctness issues (existing tables are not forward compatible), and as you said the second one would have to be a protocol change which has its own set of issues. Let's chat on Slack about how to proceed.